karma.conf.js

```diff
@@ -1,78 +1,79 @@
+    .split('\n')
+    .reduce((list, line) => {
+      // filter out node_modules
+      if (/~/.test(line)) return list
+
+      // indent the error beneath the it() message
+      let newLine = '  ' + line
 
-          // indent the error beneath the it() message
-          let newLine = '  ' + line
+      if (newLine.includes('webpack:///')) {
+        if (haveSeenStack === false) {
+          const indent = newLine.slice(0, newLine.search(/\S/))
+          newLine = `\n${indent}Stack:\n${newLine}`
+          haveSeenStack = true
+        }
 
-          if (newLine.includes('webpack:///')) {
-            if (haveSeenStack === false) {
-              const indent = newLine.slice(0, newLine.search(/\S/))
-              newLine = `\n${indent}Stack:\n${newLine}`
-              haveSeenStack = true
-            }
+        // remove webpack:///
+        newLine = newLine.replace('webpack:///', '')
 
-            // remove webpack:///
-            newLine = newLine.replace('webpack:///', '')
+        // remove bundle location, showing only the source location
+        newLine = newLine.slice(0, newLine.indexOf(' <- '))
+      }
 
-            // remove bundle location, showing only the source location
-            newLine = newLine.slice(0, newLine.indexOf(' <- '))
-          }
+      return list.concat(newLine)
+    }, [])
+    .join('\n')
+}
 
-          return list.concat(newLine)
-        }, [])
-        .join('\n')
+module.exports = (karmaConfig) => {
+  karmaConfig.set({
+    basePath: process.cwd(),
+    browsers: ['PhantomJS'],
+    client: {
+      mocha: {
+        reporter: 'html',   // change Karma's debug.html to mocha web reporter
+        ui: 'bdd',
+      },
     },
-    frameworks: [
-      'phantomjs-shim',
-      'mocha',
-    ],
     coverageReporter: {
       reporters: [
         { type: 'lcov', dir: 'coverage', subdir: '.' },
         { type: 'text-summary' },
       ],
       includeAllSources: true,
     },
+    files: [
+      './test/tests.bundle.js',
+    ],
+    formatError,
+    frameworks: ['phantomjs-shim', 'mocha'],
+    reporters: ['mocha', 'coverage'],
     preprocessors: {
       // do not include 'coverage' preprocessor for karma-coverage
       // code is already instrumented by babel-plugin-__coverage__
-      '**/*.bundle.js': ['webpack'],
-    },
-    client: {
-      mocha: {
-        reporter: 'html',   // change Karma's debug.html to mocha web reporter
-        ui: 'bdd',
-      },
+      './test/tests.bundle.js': ['webpack', 'sourcemap'],
     },
+    singleRun: !argv.watch,
     webpack: {
       devtool: config.compiler_devtool,
-      module: Object.assign({}, webpackConfig.module, {
-        loaders: [
+      entry: './test/tests.bundle.js',
+      module: {
+        rules: [
+          ...webpackConfig.module.rules,
           {
             test: /sinon\.js$/,
-            loader: 'imports?define=>false,require=>false',
+            loader: 'imports-loader?define=>false,require=>false',
```

Comment:
This is no longer required after the sinon 2 update.  We just need to rebase this branch.